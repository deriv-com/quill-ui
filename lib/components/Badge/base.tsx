import { forwardRef } from "react";
import { BadgeProps } from "./types";
import clsx from "clsx";
import "./badge.scss";
import { CaptionText, Text } from "@components/Typography";

export const BadgeBase = forwardRef<HTMLSpanElement, BadgeProps>(
    (
        { size = "sm", color = "success", label, variant, className, ...rest },
        ref,
    ) => {
        const labelColor = "badge__base-color";
        return (
            <span
                className={clsx(
                    className,
                    "badge__base",
                    `badge__variant__${variant}--size-${size}`,
                    `badge__color--${color}`,
                )}
                ref={ref}
                {...rest}
            >
                {label &&
                    (size === "sm" ? (
                        <CaptionText bold color={labelColor}>
                            {label}
                        </CaptionText>
                    ) : size === "md" ? (
                        <Text bold color={labelColor}>
                            {label}
                        </Text>
                    ) : size === "lg" ? (
                        <Text size="xl" bold color={labelColor}>
                            {label}
                        </Text>
                    ) : null)}
            </span>
        );
    },
);
export default BadgeBase;
