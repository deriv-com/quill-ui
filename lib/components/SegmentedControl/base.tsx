import React, {
    KeyboardEvent,
    ReactNode,
    RefObject,
    useEffect,
    useState,
} from "react";
import { TRegularSizes } from "@types";
import { KEY } from "@utils/common-utils";
import { Segment } from "./segment";
import "./segmented-control.scss";

export interface Option {
    disabled?: boolean;
    icon?: ReactNode;
    label?: ReactNode;
    selected?: boolean;
}

export interface SegmentedControlProps {
    className?: string;
    hasAnimation?: boolean;
    options: Array<Option>;
    onChange?: (selectedItemIndex: number) => void;
    size?: TRegularSizes;
}

export const SegmentedControl = ({
    className,
    hasAnimation,
    options = [],
    onChange,
    size = "md",
}: SegmentedControlProps) => {
    const [allowFocus, setAllowFocus] = useState(true);
    const [animatedOptionIdx, setAnimatedOptionIdx] = useState<number | null>(
        null,
    );
    const selectedRef = React.useRef<HTMLButtonElement | null>(null);

    const animate = (idx: number, ref: RefObject<HTMLButtonElement>) => {
        if (hasAnimation && ref?.current) {
            const x =
                selectedRef?.current === null
                    ? "0%"
                    : selectedRef?.current?.offsetLeft -
                      ref.current.offsetLeft +
                      "px";
            if (selectedRef?.current !== null) {
                selectedRef.current.style.removeProperty("--x");
            }
            ref.current.style.setProperty("--x", x);
            setAnimatedOptionIdx(idx);
        }
    };

    const handleKeyboardEvents = (
        e: KeyboardEvent<HTMLButtonElement>,
        idx: number,
    ) => {
        const selectedOptionIdx = options.findIndex(
            (option) => option.selected,
        );
        setAllowFocus([KEY.TAB, KEY.ENTER, KEY.SPACE].includes(e.key));
        if (!options[idx].disabled && [KEY.ENTER, KEY.SPACE].includes(e.key)) {
            onChange?.(idx);
        }
        if (e.key === KEY.ARROW_LEFT) {
            const previousEnabledOptionIdx = options.findLastIndex(
                (option, i) => i < selectedOptionIdx && !option.disabled,
            );
            if (previousEnabledOptionIdx !== -1) {
                onChange?.(previousEnabledOptionIdx);
            }
        }
        if (e.key === KEY.ARROW_RIGHT) {
            const nextEnabledOptionIdx = options.findIndex(
                (option, i) => i > selectedOptionIdx && !option.disabled,
            );
            if (nextEnabledOptionIdx !== -1) {
                onChange?.(nextEnabledOptionIdx);
            }
        }
    };

    useEffect(() => {
        let timeoutId: ReturnType<typeof setTimeout>;
        if (selectedRef?.current && hasAnimation) {
            timeoutId = setTimeout(() => {
                selectedRef.current?.style.removeProperty("--x");
            }, 150);
        }
        return () => timeoutId && clearTimeout(timeoutId);
    }, [options]);

    const segmentRef = React.useRef<HTMLButtonElement>(null);

    return (
        <div className={className}>
            {options.map(({ disabled, icon, label, selected }, idx) => {
                return (
                    <Segment
                        allowFocus={allowFocus}
                        key={`${idx}_${label}`}
                        icon={icon}
                        isAnimated={
                            selected &&
                            animatedOptionIdx === idx &&
                            hasAnimation
                        }
                        isDisabled={disabled}
                        isSelected={selected}
                        label={label}
                        onClick={() => {
                            animate(idx, segmentRef);
                            onChange?.(idx);
                            setAllowFocus(false);
                        }}
                        onKeyDown={(
                            e: React.KeyboardEvent<HTMLButtonElement>,
                        ) => handleKeyboardEvents(e, idx)}
                        size={size}
                        ref={selected ? selectedRef : segmentRef}
                    />
                );
            })}
        </div>
    );
};
