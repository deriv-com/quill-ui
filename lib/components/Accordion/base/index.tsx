import { useCallback, useEffect, useRef, useState } from "react";
import clsx from "clsx";
import { Text } from "@components/Typography";
import { StandaloneChevronDownRegularIcon } from "@deriv/quill-icons/Standalone";
import { AccordionProps } from "../types";
import "./accordion-base.scss";

export const Base = ({
    id,
    className,
    title = "",
    subtitle,
    content: Content,
    expanded = false,
    textSize = "md",
    icon,
    divider = "none",
    disabled = false,
    customContent: CustomContent,
    contentClassname,
    onExpand,
}: AccordionProps) => {
    const [isExpanded, setExpanded] = useState(expanded);
    const [isAutoExpand, setAutoExpand] = useState(false);

    const accordionElement = useRef<HTMLDivElement>(null);

    const toggleCollapse = useCallback(() => {
        setExpanded((current) => !current);
        setAutoExpand(false);
        scrollToExpanded(500);

        if (onExpand) {
            onExpand(!isExpanded, title);
        }
    }, [isExpanded, onExpand, title]);

    const handleKeyUp = useCallback(
        (e: KeyboardEvent) => {
            if (e.code === "Enter" || e.key === "Enter") {
                if (accordionElement.current === document.activeElement) {
                    toggleCollapse();
                }
            }
        },
        [toggleCollapse],
    );

    const scrollToExpanded = (delay = 1000) => {
        const accElement = accordionElement.current;

        if (accElement) {
            setTimeout(() => {
                accElement.scrollIntoView({
                    block: "center",
                    behavior: "smooth",
                });
            }, delay);
        }
    };

    useEffect(() => {
        const hashWithoutSymbol =
            typeof window !== "undefined" && window.location.hash.slice(1);

        if (hashWithoutSymbol === id) {
            setAutoExpand(true);
            scrollToExpanded();
        }
    }, [id]);

    // Key handlers
    useEffect(() => {
        document.addEventListener("keyup", handleKeyUp);
        return () => {
            document.removeEventListener("keyup", handleKeyUp);
        };
    }, [handleKeyUp]);

    useEffect(() => {
        setExpanded(expanded);
    }, [expanded]);

    return (
        <div
            data-id={id}
            ref={accordionElement}
            tabIndex={0}
            className={clsx(
                "quill-accordion-container",
                `quill-accordion-container-divider-${divider}`,
                disabled && `quill-accordion-container-${disabled}`,
                className,
                isExpanded && "quill-accordion-container-expanded-color",
            )}
        >
            <div
                className={clsx(
                    "quill-accordion-base",
                    disabled && "quill-accordion-base-disabled",
                    contentClassname,
                )}
                onClick={() => toggleCollapse()}
                data-testid="toggle-expand"
            >
                {CustomContent ? (
                    <CustomContent />
                ) : (
                    <>
                        {icon && (
                            <div
                                className={clsx(
                                    disabled && "quill-accordion-icon-disabled",
                                )}
                                data-testid="accordion-icon"
                            >
                                {icon}
                            </div>
                        )}
                        <div className={"quill-accordion-header"}>
                            <Text
                                className={clsx(
                                    "quill-accordion-base-header-title-color",
                                    `quill-accordion-base-header-title-${textSize}`,
                                    disabled &&
                                        "quill-accordion-header-title-disabled",
                                )}
                            >
                                {title}
                            </Text>
                            {subtitle && (
                                <Text
                                    className={clsx(
                                        "quill-accordion-header-subtitle-color",
                                        `quill-accordion-base-header-subtitle-${textSize}`,
                                        disabled &&
                                            "quill-accordion-header-subtitle-disabled",
                                    )}
                                >
                                    {subtitle}
                                </Text>
                            )}
                        </div>
                    </>
                )}
                <div
                    className="quill-accordion-icon-rotate"
                    data-state={isExpanded || isAutoExpand ? "open" : "close"}
                    data-testid="chevron"
                >
                    <StandaloneChevronDownRegularIcon
                        iconSize="sm"
                        fill={
                            disabled
                                ? "var(--component-accordion-icon-disabled)"
                                : "var(--component-textIcon-normal-prominent)"
                        }
                    />
                </div>
            </div>

            <div
                className={clsx(
                    disabled && "quill-accordion-disabled",
                    isAutoExpand || isExpanded
                        ? "quill-accordion-expanded"
                        : "quill-accordion-expanded-visible",
                )}
                data-testid="expanded-content"
            >
                <div className={clsx("quill-accordion-expanded-content")}>
                    {Content && <Content />}
                </div>
            </div>
        </div>
    );
};

Base.displayName = "AccordionBase";

export default Base;
