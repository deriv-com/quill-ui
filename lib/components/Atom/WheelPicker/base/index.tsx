import { useEffect, useMemo, useRef, useState } from "react";
import clsx from "clsx";
import { DropdownItem } from "@components/Atom/dropdown";
import "./wheel-picker-base.scss";
import { WheelPickerProps } from "../types";

export const WheelPicker = ({
    data,
    selectedValue,
    setSelectedValue,
    handleKeyDown,
    isFocused,
    dropDownItemClassName,
    containerClassName,
    listClassName,
    position,
    containerHeight = "100%",
    ...rest
}: WheelPickerProps) => {
    const itemsRefs = useRef<HTMLDivElement[]>([]);
    const [inputData, setInputData] = useState(data);
    const dataItemsMap = useMemo(
        () =>
            data.reduce(
                (map, item, index) => map.set(item.value, index),
                new Map(),
            ),
        [data],
    );

    const [currentDataValue, setCurrentValue] = useState(
        dataItemsMap.get(selectedValue) ?? 0,
    );
    const itemsContainerRef = useRef<HTMLUListElement>(null);

    const handleDataScroll = (event: Event) => {
        if (!event.target) return;
        const scrollTop = Math.max(
            (event.target as HTMLUListElement).scrollTop,
            0,
        );
        const selectedElement = Math.min(
            Math.max(Math.floor(scrollTop / 48), 0),
            inputData.length - 1,
        );

        (itemsRefs.current[selectedElement] as HTMLDivElement)?.focus({
            preventScroll: true,
        });
        setSelectedValue(inputData[selectedElement].value);
    };

    const resizeObserver = new ResizeObserver(() => {
        const index = Number(currentDataValue);
        if (!itemsContainerRef?.current) return;
        const divHeight = itemsContainerRef?.current.clientHeight;
        itemsContainerRef.current.style.paddingTop = `${divHeight * 0.5}px`;
        itemsContainerRef.current.style.paddingBottom = `${divHeight * 0.5}px`;
        if (itemsRefs.current[index]) {
            (itemsRefs.current[index] as HTMLDivElement).scrollIntoView({
                block: "center",
            });
        }
    });

    useEffect(() => {
        if (!itemsContainerRef.current) return;
        setSelectedValue(inputData[currentDataValue]?.value);
        resizeObserver.observe(itemsContainerRef.current);
        itemsContainerRef.current.addEventListener("scroll", handleDataScroll);
        return () => {
            resizeObserver.disconnect();
            itemsContainerRef.current?.removeEventListener(
                "scroll",
                handleDataScroll,
            );
        };
    }, [inputData]);

    useEffect(() => {
        setInputData(data);
        setCurrentValue(dataItemsMap.get(selectedValue) ?? 0);
    }, [data]);

    useEffect(() => {
        if (isFocused) {
            itemsRefs.current[dataItemsMap.get(selectedValue)]?.focus();
        }
    }, [isFocused]);

    return (
        <div
            className={clsx(
                "quill-wheel-picker__container",
                `quill-wheel-picker__container-disabled-${rest?.disabled}`,
                containerClassName,
            )}
            style={{ height: containerHeight }}
        >
            <ul
                className={clsx(
                    "quill-wheel-picker__data-items",
                    `quill-wheel-picker__data-items-${position}`,
                    listClassName,
                )}
                role="listbox"
                ref={itemsContainerRef}
                {...rest}
            >
                {inputData.map(({ value, label }, index) => {
                    return (
                        <DropdownItem
                            className={clsx(
                                "quill-wheel-picker__data-item",
                                dropDownItemClassName,
                            )}
                            tabIndex={0}
                            label={label ?? value}
                            role="option"
                            textAlignment="center"
                            ref={(node: HTMLDivElement) =>
                                (itemsRefs.current[index] = node)
                            }
                            key={value}
                            onKeyDown={handleKeyDown}
                            onClick={() => {
                                itemsRefs.current[index]?.scrollIntoView({
                                    block: "center",
                                    behavior: "smooth",
                                    inline: "nearest",
                                });
                                itemsRefs.current[index]?.focus();
                            }}
                            disabled={inputData[index].value !== selectedValue}
                        >
                            {value}
                        </DropdownItem>
                    );
                })}
            </ul>
        </div>
    );
};

export default WheelPicker;
