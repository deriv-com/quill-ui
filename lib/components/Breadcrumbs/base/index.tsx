import { BreadcrumbProps, LinkProps } from "../types";
import { Fragment, useEffect, useState } from "react";
import { maxLinks } from "./constants";
import "./breadcrumbs.scss";
import useBreakpoints from "@hooks/useBreakpoints";
import clsx from "clsx";
import Link from "@components/Link";

export function Base({ size = "sm", links = [], className }: BreadcrumbProps) {
    const [renderLinks, setRenderLinks] = useState<LinkProps[]>([]);
    const [dropdownLinks, setDropdownLinks] = useState<LinkProps[]>([]);

    const isLastItem = (key: number) => renderLinks.length - 1 > key;
    const linksLen = links.length;
    const { isMobile } = useBreakpoints();
    const hasExtra = isMobile && maxLinks < linksLen;

    useEffect(() => {
        setRenderLinks(links);

        if (hasExtra) {
            const limitedLinks = [
                links[0],
                { content: "...", href: "" },
                links[linksLen - 2],
                links[linksLen - 1],
            ];

            setRenderLinks(limitedLinks);

            const remainingLinks: LinkProps[] = links.filter(
                (item) => !new Set(limitedLinks).has(item),
            );

            setDropdownLinks(remainingLinks);
        }
    }, [links]);

    return (
        <div className={clsx("breadcrumb--container", className)}>
            {renderLinks.map(({ content, href }, lk) => (
                <Fragment key={`breadcrumbs-${content}`}>
                    {hasExtra && lk === 1 && (
                        <select
                            className="breadcrumb--dropdown"
                            onChange={(
                                e: React.ChangeEvent<HTMLSelectElement>,
                            ) => {
                                if (typeof window !== "undefined") {
                                    const url = e.target.value;

                                    try {
                                        const sanitizedUrl = new URL(url);
                                        window.location.href =
                                            sanitizedUrl.href;
                                    } catch (err) {
                                        console.error("Invalid URL:", url);
                                    }
                                }
                            }}
                        >
                            {dropdownLinks.map(({ href, content }) => {
                                return (
                                    <option
                                        key={`dropdown-${content}`}
                                        value={href}
                                    >
                                        {content}
                                    </option>
                                );
                            })}
                        </select>
                    )}

                    <Link
                        size={size}
                        href={href}
                        hasHoverEffect={false}
                        hasChevron={isLastItem(lk)}
                        color="black"
                        disabled={!isLastItem(lk)}
                        className="breadcrumb--item"
                    >
                        {content}
                    </Link>
                </Fragment>
            ))}
        </div>
    );
}

export default Base;
