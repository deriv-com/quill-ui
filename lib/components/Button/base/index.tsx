import { forwardRef } from "react";
import clsx from "clsx";
import { ButtonProps } from "../types";
import {
    LabelPairedChevronDownLgBoldIcon,
    LabelPairedChevronDownMdBoldIcon,
    LabelPairedChevronDownSmBoldIcon,
    LabelPairedChevronDownXlBoldIcon,
    LabelPairedLoaderCaptionRegularIcon,
    LabelPairedLoaderLgRegularIcon,
    LabelPairedLoaderMdRegularIcon,
    LabelPairedLoaderSmRegularIcon,
} from "@deriv/quill-icons";
import "../button.scss";
import { CaptionText, Text } from "@components/Typography";

export const ButtonSize = {
    xl: "quill-button__size--xl",
    lg: "quill-button__size--lg",
    md: "quill-button__size--md",
    sm: "quill-button__size--sm",
} as const;

const dropdownIcons = {
    sm: LabelPairedChevronDownSmBoldIcon,
    md: LabelPairedChevronDownMdBoldIcon,
    lg: LabelPairedChevronDownLgBoldIcon,
    xl: LabelPairedChevronDownXlBoldIcon,
};
export const loaderIcons = {
    sm: LabelPairedLoaderCaptionRegularIcon,
    md: LabelPairedLoaderSmRegularIcon,
    lg: LabelPairedLoaderMdRegularIcon,
    xl: LabelPairedLoaderLgRegularIcon,
};

export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
    (
        {
            className,
            color = "coral",
            icon,
            children,
            isDropdownOpen,
            dropdown = false,
            selected,
            fullWidth = false,
            isLoading = false,
            size = "md",
            label,
            disabled,
            iconPosition,
            variant = "primary",
            ...rest
        },
        ref,
    ) => {
        const buttonColorClass = `quill__color--${variant}-${color}`;
        const labelSize = size === "md" ? "sm" : size === "lg" ? "md" : "xl";
        const DropdownIcon = dropdownIcons[size];
        const LoaderIcon = loaderIcons[size];

        return (
            <button
                className={clsx(
                    "quill-button",
                    ButtonSize[size],
                    buttonColorClass,
                    className,
                    fullWidth && "quill-button__full-width",
                )}
                disabled={disabled}
                data-state={selected ? "selected" : ""}
                ref={ref}
                {...rest}
            >
                {iconPosition === "start" && icon && !isLoading && icon}

                {isLoading && !disabled && (
                    <LoaderIcon
                        data-testid="button-loader"
                        className="quill-button__loader-icon"
                    />
                )}
                {children && <div>{children}</div>}
                {label && !isLoading && (
                    <span className="quill-button-label">
                        {size === "sm" ? (
                            <CaptionText color={color} bold>
                                {label}
                            </CaptionText>
                        ) : (
                            <Text size={labelSize} bold color={color}>
                                {label}
                            </Text>
                        )}
                    </span>
                )}
                {iconPosition === "end" && icon && !isLoading && icon}
                {dropdown && DropdownIcon && !isLoading && (
                    <DropdownIcon
                        data-state={isDropdownOpen ? "open" : "close"}
                        className={clsx(
                            "quill-button__transform",
                            isDropdownOpen && "quill-button__transform-rotate",
                        )}
                    />
                )}
            </button>
        );
    },
);

Button.displayName = "Button";

export default Button;
