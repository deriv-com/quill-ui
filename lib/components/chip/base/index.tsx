import React, { useState } from "react";
import { SelectableChipProps } from "../types";
import { StandardSizes } from "../../../types";
import {
    StandaloneChevronDownRegularIcon,
    StandaloneCircleXmarkRegularIcon,
} from "@deriv/quill-icons";
import "../_chip.scss";
import clsx from "clsx";

export const ChipIconSizes: Record<
    StandardSizes,
    { width: number; height: number }
> = {
    sm: {
        width: 11,
        height: 18,
    },
    md: {
        width: 13,
        height: 22,
    },
    lg: {
        width: 14,
        height: 24,
    },
};

export const Base = ({
    icon: Icon,
    size = "md",
    children,
    labelTag,
    dismissible = false,
    dropdown = false,
    className,
    onChipSelect,
    onDismiss,
    ...rest
}: SelectableChipProps) => {
    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
        // if (dismissible || dropdown) return;
        const target = event.currentTarget;
        const icon = target.querySelector(".rotate");
        const isOpen = icon?.getAttribute("data-state") === "open";
        const selected_state = isOpen ? "close" : "open";
        icon?.setAttribute("data-state", selected_state);
        onChipSelect?.(event, !isOpen);
    };

    const handleDismiss = (
        event: React.MouseEvent<SVGSVGElement, MouseEvent>,
    ) => {
        onDismiss?.(event);
    };

    return (
        <button
            onClick={handleClick}
            className={clsx(`quill-chip__size--${size}`, className)}
            {...rest}
        >
            {Icon && <Icon {...ChipIconSizes[size]} />}
            {children}
            {labelTag && <span className="">{labelTag}</span>}
            {dismissible && (
                <StandaloneCircleXmarkRegularIcon
                    {...ChipIconSizes[size]}
                    onClick={handleDismiss}
                    data-testid="dt-chip-dismissible-btn"
                    className="cursor-pointer"
                />
            )}
            {dropdown && (
                <StandaloneChevronDownRegularIcon
                    width={24}
                    height={24}
                    data-state="open"
                    className="rotate"
                />
            )}
        </button>
    );
};

export default Base;
