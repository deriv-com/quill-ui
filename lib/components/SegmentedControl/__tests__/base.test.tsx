import React from "react";
import { render, fireEvent, act } from "@testing-library/react";
import { SegmentedControl, SegmentedControlProps } from "../base";

const defaultProps: SegmentedControlProps = {
    options: [
        { label: "Option 1", selected: true },
        { label: "Option 2" },
        { label: "Option 3", disabled: true },
    ],
    onChange: jest.fn(),
    hasAnimation: false,
};

describe("SegmentedControl", () => {
    it("renders correctly and matches snapshot", () => {
        const { container } = render(<SegmentedControl {...defaultProps} />);
        expect(container).toMatchSnapshot();
    });

    it("calls onChange with the correct index when an option is clicked", () => {
        const onChangeMock = jest.fn();
        const { getByText } = render(
            <SegmentedControl {...defaultProps} onChange={onChangeMock} />,
        );

        act(() => {
            fireEvent.click(getByText("Option 2"));
        });
        expect(onChangeMock).toHaveBeenCalledWith(1);
    });

    it("does not call onChange when a disabled option is clicked", () => {
        const onChangeMock = jest.fn();
        const { getByText } = render(
            <SegmentedControl {...defaultProps} onChange={onChangeMock} />,
        );

        act(() => {
            fireEvent.click(getByText("Option 3"));
        });
        expect(onChangeMock).not.toHaveBeenCalled();
    });

    it("renders with animation when hasAnimation is true", () => {
        const animatedProps = { ...defaultProps, hasAnimation: true };
        const { container } = render(<SegmentedControl {...animatedProps} />);
        expect(container).toMatchSnapshot();
    });
});
