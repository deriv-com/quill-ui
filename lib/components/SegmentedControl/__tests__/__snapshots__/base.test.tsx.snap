// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SegmentedControl renders correctly and matches snapshot 1`] = `
<div>
  <div>
    <button
      class="item selected"
    >
      <span
        class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Option 1
      </span>
    </button>
    <button
      class="item"
    >
      <span
        class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Option 2
      </span>
    </button>
    <button
      class="item"
      disabled=""
    >
      <span
        class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Option 3
      </span>
    </button>
  </div>
</div>
`;

exports[`SegmentedControl renders with animation when hasAnimation is true 1`] = `
<div>
  <div>
    <button
      class="item selected"
    >
      <span
        class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Option 1
      </span>
    </button>
    <button
      class="item"
    >
      <span
        class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Option 2
      </span>
    </button>
    <button
      class="item"
      disabled=""
    >
      <span
        class="quill-typography quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default"
      >
        Option 3
      </span>
    </button>
  </div>
</div>
`;
